frontend/lib/parsing/ParserContext.h

```diff
@@ -274,6 +280,18 @@ struct ParserContext {
 
   Location convertLocation(YYLTYPE location);
 
+  WithClause* buildWithClause(YYLTYPE location, YYLTYPE locWith,
+                              YYLTYPE locLeftParen,
+                              YYLTYPE locTaskIntentList,
+                              YYLTYPE locRightParen,
+                              ParserExprList* exprList);
+
+  CommentsAndStmt buildBeginStmt(YYLTYPE location, YYLTYPE locBegin,
+                                 YYLTYPE locWithClause,
+                                 YYLTYPE locStmt,
+                                 WithClause* withClause,
+                                 CommentsAndStmt stmt);
+
```

Comment:
Why did you need to change these for `Dot` fields and `NamedDecl`s?

Reply:
In general, I think it's the right thing to do to pull any parser action more than one call long out into a separate builder. This is something we didn't always do when we rushed to get the parser in, but I think it's the right thing to do now if your work needs to touch a particular multi-line action. I pulled these out because I had to correct them to fix bugs.